tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/TimeRange.java

```diff
@@ -0,0 +1,279 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.tsfile.read.common;
+
+import java.util.ArrayList;
+import org.apache.iotdb.tsfile.read.expression.IExpression;
+import org.apache.iotdb.tsfile.read.expression.impl.BinaryExpression;
+import org.apache.iotdb.tsfile.read.expression.impl.GlobalTimeExpression;
+import org.apache.iotdb.tsfile.read.filter.TimeFilter;
+
+/**
+ * interval [min,max] of long data type
+ *
+ * Reference: http://www.java2s.com/Code/Java/Collections-Data-Structure/Anumericalinterval.htm
+ *
+ * @author ryanm
+ */
+public class TimeRange implements Comparable<TimeRange> {
+
+  /**
+   * The lower value
+   */
+  private long min = 0;
+
+  /**
+   * The upper value
+   */
+  private long max = 0;
+
+  /**
+   * @param min
+   * @param max
+   */
+  public TimeRange(long min, long max) {
+    set(min, max);
+  }
+
+  public int compareTo(TimeRange r) {
```

Comment:
Do we need @Override above the compareTo()  method ?
it is recommended that we override equals() method as well